### PR TITLE
Refactor GaussianProcessClassification_unittest.cc to use fixtures

### PR DESCRIPTION
**Pull request by durovo**
This PR refactors GaussianProcessClassification_unittest.cc to use fixtures. (See issue #3947 ).

### CODE REVIEW

File: tests/unit/classifier/GaussianProcessClassification_unittest.cc [hunk 1]
```diff
@@ -67,68 +67,402 @@
...
+		GaussianProcessClassification::TearDown();
+
+		SG_UNREF(gpc);
+	}
+	float64_t abs_tolerance;
+	const float64_t rel_tolerance = 1e-2;
+	CGaussianProcessClassification* gpc;
+};
+
+// empty class since TEST and TEST_F cannot be used in the same file
```
Comment (karlnapf) [inline tests/unit/classifier/GaussianProcessClassification_unittest.cc]: I am not sure I understand what you mean here. I definitely had `TEST` and `TEST_F` in the same file before ...
Also not very pretty :/

Comment (durovo) [reply tests/unit/classifier/GaussianProcessClassification_unittest.cc]: You are right.. I was under the impression that TEST and TEST_F cannot be used in the same file. Should I remove the empty class and change the corresponding `TEST_F`s to `TEST`s?

**Comment by durovo** [comment]
I have removed the empty class and refactored the corresponding `TEST_F`s to `TEST`s.

**Comment by karlnapf** [comment]
Well done on your first Shogun patch! :)
Any plans for next steps?

**Comment by durovo** [comment]
Thanks a lot :). I am planning to work on #3851 next.

**Pull request merged**